src/components/GraphFilter/GraphFind.tsx

```diff
@@ -47,21 +51,28 @@ export class GraphFind extends React.PureComponent<GraphFindProps> {
       props.toggleFindHelp();
     }
 
-    this.hideInputRef = React.createRef();
-    this.hideInputValue = '';
-    this.hideValue = '';
-    this.hiddenElements = undefined;
-
     this.findInputRef = React.createRef();
     this.findInputValue = '';
-    this.findValue = '';
+
+    this.hiddenElements = undefined;
+
+    this.hideInputRef = React.createRef();
+    this.hideInputValue = '';
   }
 
   componentDidUpdate(prevProps: GraphFindProps) {
-    if (this.findValue.length > 0 && this.props.cyData.updateTimestamp !== prevProps.cyData.updateTimestamp) {
+    // make sure the input box reflects the redux value
+    this.findInputRef.value = this.props.findValue;
+    this.hideInputRef.value = this.props.hideValue;
+
+    const findChanged = this.props.findValue !== prevProps.findValue;
+    const hideChanged = this.props.hideValue !== prevProps.hideValue;
+    const graphChanged = prevProps.cyData && this.props.cyData.updateTimestamp !== prevProps.cyData.updateTimestamp;
```

Comment:
I'm curious why you tested from `prevProps.cyData` but not for `this.props.cyData`
Are we sure that we will never hit an update with cyData = null?

Reply:
Because this is in ComponentDidUpdate, which is called after the first render, I think it should exist.  prevProps doesn't exist until the second time.  If you think it's unsafe I can add a check.

Reply:
`componentDidUpdate()` is called on the first re-render, prevProps should already exist there. I'm not sure if `cyData` is always present, i asked out of curiosity to see if you knew about it.

Reply:
I will add the check.